tests/template/extension_test.php

```diff
@@ -253,8 +259,197 @@ public function data_template_extensions()
 	/**
 	 * @dataProvider data_template_extensions
 	 */
-	public function test_get_user_avatar($file, $vars, $block_vars, $destroy_array, $expected, $lang_vars = [])
+	public function test_template_extensions($file, $vars, $block_vars, $destroy_array, $expected, $lang_vars = [])
 	{
 		$this->run_template($file, $vars, $block_vars, $destroy_array, $expected, $lang_vars);
 	}
+
+	public function data_template_icon_extension()
+	{
+		return [
+			/** Font: default */
+			[
+				[
+					'type'			=> 'font',
+					'icon'			=> 'phone',
+					'title'			=> 'ICON_PHONE',
+					'hidden'		=> false,
+					'classes'		=> '',
+					'attributes'	=> [],
+				],
+				[
+					'ICON_PHONE'	=> 'Phone icon',
+				],
+				'<i class="o-icon font fa-phone"></i><span>Phone icon</span>',
+
+			],
+			/** Font: all options */
+			[
+				[
+					'type'			=> 'font',
+					'icon'			=> 'pencil',
+					'title'			=> 'ICON_PENCIL',
+					'hidden'		=> true,
+					'classes'		=> 'a-class another-class',
+					'attributes'	=> [
+						'data-attr-1'	=> 'true',
+						'data-attr-2'	=> 'two',
+					],
+				],
+				[
+					'ICON_PENCIL'	=> 'Pencil icon',
+				],
+				'<i class="o-icon font fa-pencil a-class another-class" title="Pencil icon" aria-hidden="true" data-attr-1="true" data-attr-2="two"></i>
+				<span class="sr-only">Pencil icon</span>'
+			],
+			/** Iconify: default */
+			[
+				[
+					'type'			=> 'iconify',
+					'icon'			=> 'phone',
+					'title'			=> '',
+					'hidden'		=> false,
+					'classes'		=> '',
+					'attributes'	=> [],
+				],
+				[],
+				'<i class="iconify o-icon phone"></i>',
+			],
+			/** Iconify: all options */
+			[
+				[
+					'type'			=> 'iconify',
+					'icon'			=> 'pencil',
+					'title'			=> 'ICON_PENCIL',
+					'hidden'		=> true,
+					'classes'		=> 'icon-lg',
+					'attributes'	=> [
+						'data-swap'		=> 'Swap text',
+					],
+				],
+				[
+					'ICON_PENCIL'	=> 'Pencil icon',
+				],
+				'<i class="iconify o-icon pencil icon-lg" title="Pencil icon" aria-hidden="true" data-swap="Swap text"></i>
```

Comment:
update to `'<i class="iconify o-icon icon-lg" title="Pencil icon" aria-hidden="true" data-icon="fa:pencil" data-inline="true" data-swap="Swap text"></i>`